### PR TITLE
fixed dynamic shape in customer model

### PR DESCRIPTION
**Pull request by chxin66**
Signed-off-by: Chen Xin <jack.chen@verisilicon.com>

### CODE REVIEW

File: op_map.cc [hunk 1]
```diff
@@ -689,12 +688,54 @@ struct SimpleOpWithFusedActiovationMapper
...
+      }
+      std::vector<int32_t> in_1(total_input);
+      assert(inputs[1]->CopyDataFromTensor(in_1.data()));
+
+      for (int i = 0; i < outputs[0]->GetShape().size(); ++i) {
+        total_output *= outputs[0]->GetShape()[i];
+      }
+      std::vector<int32_t> out(total_output);
+
+      tim::vx::TensorSpec output_spec(tim::vx::DataType::INT32,
```
Comment (sunshinemyson) [inline op_map.cc:714]: output always int32?

File: op_map.cc [hunk 2]
```diff
@@ -889,7 +930,7 @@ struct TransposeConvMapper : public OpMapperBase<TfLiteTransposeConvParams> {
 
     std::vector<int32_t> output_shape(inputs[0]->GetShape()[0]);
     inputs[0]->CopyDataFromTensor(output_shape.data());
-
+    // output_shape[1]=1; //按照pack的输出output_shape[1]是8，hack为1后正确
```
Comment (sunshinemyson) [inline op_map.cc:933]: Fix this ?

File: op_map.cc [hunk 3]
```diff
@@ -1300,18 +1341,56 @@ struct StridedSliceMapper : public OpMapperBase<TfLiteStridedSliceParams> {
...
+        total_input *= inputs[0]->GetShape()[i];
+      }
+      for (int i = 0; i < outputs[0]->GetShape().size(); ++i) {
+        total_output *= outputs[0]->GetShape()[i];
+      }
+      std::vector<int32_t> in(total_input);
+      std::vector<int32_t> out(total_output);
+      assert(inputs[0]->CopyDataFromTensor(in.data()));
+
+      tim::vx::TensorSpec output_spec(tim::vx::DataType::INT32,
```
Comment (sunshinemyson) [inline op_map.cc:1359]: always int32?

File: op_map.cc [hunk 4]
```diff
@@ -2553,13 +2621,60 @@ struct PackMapper : public OpMapperBase<TfLitePackParams> {
...
+      std::vector<int32_t> in1(total_input);
+      std::vector<int32_t> in2(total_input);
+      std::vector<int32_t> in3(total_input);
+      std::vector<int32_t> out(total_output);
+      assert(inputs[0]->CopyDataFromTensor(in.data()));
+      assert(inputs[1]->CopyDataFromTensor(in1.data()));
+      assert(inputs[2]->CopyDataFromTensor(in2.data()));
+      assert(inputs[3]->CopyDataFromTensor(in3.data()));
+
+      tim::vx::TensorSpec output_spec(tim::vx::DataType::INT32,
```
Comment (sunshinemyson) [inline op_map.cc:2652]: int32?

**Review by sunshinemyson — CHANGES_REQUESTED** [review]

File: op_map.cc [hunk 5]
```diff
@@ -397,6 +397,32 @@ struct OpMapperBase : public vx::op_map::IOpMapper {
 namespace vx {
 namespace op_map {
 
+std::vector<uint8_t> read_output_data(std::shared_ptr<tim::vx::Tensor> tensor) {
```
Comment (sunshinemyson) [inline op_map.cc:400]: bool read_data(const .. & tensor, std::vector<uint8_t>& out_data)

**Review by sunshinemyson — CHANGES_REQUESTED** [review]

File: op_map.cc (hunk 5 above)
Comment (chxin66) [reply op_map.cc]: done

**Pull request closed without merging**